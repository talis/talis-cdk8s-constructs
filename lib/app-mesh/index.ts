export * from "./ingress-gateway-nlb";
export * from "./app-service";
