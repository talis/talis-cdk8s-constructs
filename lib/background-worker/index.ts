export * from "./background-worker-props";
export * from "./background-worker";
