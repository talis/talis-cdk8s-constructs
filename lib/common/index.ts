export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
