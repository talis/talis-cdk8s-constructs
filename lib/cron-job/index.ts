export * from "./cron-job-props";
export * from "./cron-job";
