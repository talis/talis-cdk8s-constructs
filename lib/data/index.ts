export * from "./config-map";
export * from "./secret";
export * from "./factory";
