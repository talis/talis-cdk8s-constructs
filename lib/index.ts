export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
