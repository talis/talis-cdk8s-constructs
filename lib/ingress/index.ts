export * from "./ingress-props";
export * from "./ingress";
