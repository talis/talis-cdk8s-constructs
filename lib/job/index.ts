export * from "./job-props";
export * from "./job";
