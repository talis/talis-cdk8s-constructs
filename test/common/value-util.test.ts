import { IntOrString } from "../../imports/k8s";
import { ensureArray, getValueFromIntOrPercent } from "../../lib";

describe("value-util", () => {
  describe("ensureArray", () => {
    test("returns an array from a string", () => {
      expect(ensureArray("foo")).toEqual(["foo"]);
    });

    test("returns an array from an array of strings", () => {
      expect(ensureArray(["foo", "bar"])).toEqual(["foo", "bar"]);
    });

    test("returns an array from a number", () => {
      expect(ensureArray(1)).toEqual([1]);
    });

    test("returns an array from an array of numbers", () => {
      expect(ensureArray([1, 2])).toEqual([1, 2]);
    });
  });

  describe("getValueFromIntOrPercent", () => {
    test.each([
      [IntOrString.fromNumber(1), 99, 1],
      [IntOrString.fromNumber(2), 99, 2],
    ])("returns a number from a number %o", (intOrStr, reference, expected) => {
      expect(getValueFromIntOrPercent(intOrStr, reference)).toEqual(expected);
    });

    test.each([
      [IntOrString.fromString("1%"), 2, 1],
      [IntOrString.fromString("33%"), 3, 1],
      [IntOrString.fromString("34%"), 3, 2],
      [IntOrString.fromString("50%"), 5, 3],
    ])(
      "returns a number from a percentage string %o, rounded up",
      (value, total, expected) => {
        expect(getValueFromIntOrPercent(value, total)).toEqual(expected);
      },
    );

    test.each([
      [IntOrString.fromString("1%"), 2, 0],
      [IntOrString.fromString("33%"), 3, 0],
      [IntOrString.fromString("34%"), 3, 1],
      [IntOrString.fromString("50%"), 5, 2],
    ])(
      "returns a number from a percentage string %o, rounded down",
      (value, total, expected) => {
        expect(getValueFromIntOrPercent(value, total, false)).toEqual(expected);
      },
    );

    test("throws if string does not have % suffic", () => {
      expect(() => {
        getValueFromIntOrPercent(IntOrString.fromString("42"), 42);
      }).toThrowErrorMatchingSnapshot();
    });

    test("throws if string is not a percentage number", () => {
      expect(() => {
        getValueFromIntOrPercent(IntOrString.fromString("foo%"), 42);
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
