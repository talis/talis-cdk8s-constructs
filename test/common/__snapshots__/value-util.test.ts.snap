// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`value-util > getValueFromIntOrPercent > throws if string does not have % suffic 1`] = `"Invalid type: 42 is not a percentage"`;

exports[`value-util > getValueFromIntOrPercent > throws if string is not a percentage number 1`] = `"Invalid value: foo% is not a number"`;
